### ISSUE TITLE
Rule Change: (no-undef)

### ISSUE THREAD

**Issue by thelovekesh** [eslint/eslint#17216]
### What rule do you want to change?

no-undef

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
const obj = {
	a: 1,
};

({ a } = obj);

console.log(a);
```

### What does the rule currently do for this code?

It shows error as:

```md
'a' is not defined (no-undef)
```

### What will the rule do after it's changed?

It should not show an error when using an object literal destructuring assignment without a declaration.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Please check the [ref](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#assignment_separate_from_declaration_2:~:text=Note%3A%20The%20parentheses%20(%20...%20)%20around%20the%20assignment%20statement%20are%20required%20when%20using%20object%20literal%20destructuring%20assignment%20without%20a%20declaration.) on MDN.

**Comment by Rec0iL99** [comment]
Hi @thelovekesh, thanks for the issue. This is the main point of using the `no-undef` rule ([Disallow the use of undeclared variables](https://eslint.org/docs/latest/rules/no-undef)).

duplicate of #12449.
